vraptor-core/src/main/java/br/com/caelum/vraptor/observer/ExecuteMethod.java

```diff
@@ -79,6 +85,11 @@ public Void call() throws Exception {
 
 				log.debug("Invoking {}", reflectionMethod);
 				Object instance = event.getControllerInstance();
+				
+				Class<?> declaringClass = reflectionMethod.getDeclaringClass();
+				if (!declaringClass.isAssignableFrom(instance.getClass())) {
+					instance = container.instanceFor(declaringClass);
```

Comment:
the problem is that AFAIK we dont do a real forward, we just execute the new controller 
method... so in the `ExecuteMethod` the `controllerInstance` still uses the first instance. 
The weird think is that it works when we forward programmatically, just like this:
```
result.forwardTo(this).other();
```
but in that case, Mirror works automagically, even with the wrong class :) bug or behavior?